File: src/main/java/com/patrickgrimard/examples/ItemRepository.java
package com.patrickgrimard.examples;

import org.springframework.data.repository.CrudRepository;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;

/**
 * Created on 2016-11-25
 *
 * @author Patrick
 */
@RepositoryRestResource
public interface ItemRepository extends CrudRepository<Item, String> {
}

File: src/main/java/com/patrickgrimard/examples/ItemRepository.kt
package com.patrickgrimard.examples

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.data.rest.core.annotation.RepositoryRestResource

/**
 *
 *
 * Created on 2016-12-03
 *
 * @author Patrick
 */
@RepositoryRestResource
interface ItemRepository : JpaRepository<Item, String>, JpaSpecificationExecutor<Item>